staging/src/k8s.io/client-go/util/certificate/csr/csr.go

```diff
@@ -150,6 +150,9 @@ func ensureCompatible(new, orig *certificates.CertificateSigningRequest, private
 	if !reflect.DeepEqual(newCSR.Subject, origCSR.Subject) {
 		return fmt.Errorf("csr subjects differ: new: %#v, orig: %#v", newCSR.Subject, origCSR.Subject)
 	}
+	if new.Spec.SignerName != nil && orig.Spec.SignerName != nil && *new.Spec.SignerName != *orig.Spec.SignerName {
```

Comment:
It's worth noting that `RequestCertificate` accepts `signerName` as a `string`, which means that `orig` can *never* be `nil` here. @liggitt was it a mistake that I made signerName (the arg) a non-ptr? Is it ever desirable to allow people to mark it as nil and defer the decision to the apiserver?

Reply:
> was it a mistake that I made signerName (the arg) a non-ptr

No, making callers understand and choose their signer matches my expectations and the likely shape of the v1 API.